op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,278 @@
+package ws
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/gorilla/websocket"
+)
+
+// FlashblockHandler manages WebSocket connections for flashblocks
+type FlashblockHandler interface {
+	// Start initializes and starts the flashblocks handler
+	Start(ctx context.Context) error
+	// Stop closes any open WebSocket connections
+	Stop()
+	// BroadcastMessage sends a message to all connected WebSocket clients
+	BroadcastMessage(message []byte)
+}
+
+// Config contains configuration for the flashblocks handler
+type Config struct {
+	// WebsocketServerPort is the port to listen for WebSocket connections
+	WebsocketServerPort int
+	// RollupBoostWsURL is the URL of the rollup boost WebSocket
+	RollupBoostWsURL string
+}
+
+// Handler implements the FlashblockHandler interface
+type Handler struct {
+	cfg             Config
+	log             log.Logger
+	isLeaderFn      func(context.Context) bool
+	wsClientMu      sync.Mutex
+	rollupBoostConn *websocket.Conn
+	server          *http.Server
+	wsClients       []*websocket.Conn
+}
+
+// NewHandler creates a new flashblocks handler
+func NewHandler(cfg Config, log log.Logger, isLeaderFn func(context.Context) bool) FlashblockHandler {
+	return &Handler{
+		cfg:        cfg,
+		log:        log,
+		isLeaderFn: isLeaderFn,
+	}
+}
+
+// Start initializes the flashblocks handler and starts the WebSocket server and rollup boost listener
+func (h *Handler) Start(ctx context.Context) error {
+	// Start the WebSocket server
+	if err := h.startWebSocketServer(ctx); err != nil {
+		return err
+	}
+
+	// Start the rollup boost listener
+	if err := h.startRollupBoostListener(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Stop closes any open WebSocket connections and shuts down the server
+// This should only be called if the conductor is shut down
+func (h *Handler) Stop() {
+	// Close the rollup boost connection if it exists
+	if h.rollupBoostConn != nil {
+		h.rollupBoostConn.Close()
+		h.rollupBoostConn = nil
+	}
+
+	// Close all WebSocket client connections
+	h.wsClientMu.Lock()
+	for _, client := range h.wsClients {
+		client.Close()
+	}
+	h.wsClients = nil
+	h.wsClientMu.Unlock()
+
+	// Force close the HTTP server if it's running
+	if h.server != nil {
+		h.log.Info("Forcibly closing WebSocket server")
+		err := h.server.Close()
+		if err != nil {
+			h.log.Error("Error closing WebSocket server", "err", err)
+		}
+		h.log.Info("WebSocket server closed")
+	}
+}
+
+// BroadcastMessage sends a message to all connected WebSocket clients
+func (h *Handler) BroadcastMessage(message []byte) {
+	h.wsClientMu.Lock()
+	defer h.wsClientMu.Unlock()
```

Comment:
I don't think we need a lock here? Feels fine if we miss a connection during addition / deletion